src/iterators/mod.rs

```diff
@@ -47,6 +47,19 @@ pub struct HistogramIterator<'a, T: 'a + Counter, P: PickyIterator<T>> {
     picker: P,
 }
 
+/// The value emitted at each step when iterating over a `Histogram`.
+#[derive(Debug, PartialEq)]
+pub struct IterationValue<T: Counter> {
+    /// the lowest value stored in the current histogram bin
+    pub value: u64,
+    /// percent of recorded values that are equivalent to or below `value`
+    pub percentile: f64,
+    /// recorded count for values equivalent to `value`
+    pub count_at_value: T,
+    /// number of values traversed since the last iteration step
+    pub count_since_last_iteration: u64
+}
+
```

Comment:
I think I'd prefer if there weren't `pub`, and instead there were `&self` accessors. We never expect an `IterationValue` to have its internals modified. We'll also need an `impl` block anyway for `::new`.